src/Compilers/CSharp/Portable/Emitter/Model/AttributeDataAdapter.cs

```diff
@@ -108,6 +109,9 @@ private Cci.IMetadataExpression CreateMetadataExpression(TypedConstant argument,
                 case TypedConstantKind.Array:
                     return CreateMetadataArray(argument, context);
 
+                case TypedConstantKind.Primitive when argument.TypeInternal.SpecialType is SpecialType.System_String && argument.ValueInternal is TypeSymbol:
```

Comment:
>argument.ValueInternal is TypeSymbol [](http://example.com/codeflow?start=120&length=36)

```TypedConstant``` is a public API. Are we certain we never expose instances like this to the public? Otherwise, ```Value``` has unexpected type (not a string).

Reply:
Tracing the usages, I couldn't find any way this weird `TypedConstant` could be seen outside the compiler.
It is synthesized in `AddSynthesizedReturnTypeAttributes` (internal) which is called as part of CCI API (Cci.IMethodDefinition.GetReturnValueAttributes, internal). Tracing the usages further we stay within CCI (writing metadata tables, indexing references) which also don't expose the attribute/constant publicly.